src/Routing/RouteSubscriber.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Drupal\tide_landing_page\Routing;
+
+use Drupal\Core\Routing\RouteSubscriberBase;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Listens to the dynamic route events.
+ */
+class RouteSubscriber extends RouteSubscriberBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+    if ($route = $collection->get('block.admin_display')) {
+      $route->setRequirement('_custom_access', 'Drupal\tide_landing_page\Access\BlockAccessCheck::access');
```

Comment:
How does use access `block/add` page?

Reply:
I have added a custom menu item (Custom block library) under structure to allow users to create/manage blocks.

Reply:
based on your changes, administrator user will have `block layout` and its sub menus(`Add custom block`, `Custom block library`), and your defined `Custom block library` which seems not right.

Reply:
@vincent-gao , I have updated the PR description. Please read above.